iota-client/src/client.rs

```diff
@@ -108,37 +116,72 @@ impl std::fmt::Debug for Client {
     }
 }
 
+impl Drop for Client {
+    /// Gracefully shutdown the `Client`
+    fn drop(&mut self) {
+        self.stop_sync.store(true, Ordering::Relaxed);
+        if let Some(handle) = self.sync_handle.take() {
+            handle.join().expect("failed to join node sync thread");
+        }
+    }
+}
+
 impl Client {
     /// Create the builder to instntiate the IOTA Client.
     pub fn builder() -> ClientBuilder {
         ClientBuilder::new()
     }
 
-    // TODO Implement syncing process
+    /// Sync the node lists per node_sync_interval milliseconds
+    pub(crate) fn start_sync_process(
+        mut runtime: Runtime,
+        sync: Arc<RwLock<HashSet<Url>>>,
+        nodes: Vec<Url>,
+        node_sync_interval: NonZeroU64,
+        stop_sync: Arc<AtomicBool>,
+    ) -> thread::JoinHandle<()> {
+        let node_sync_interval = Duration::from_millis(node_sync_interval.into());
+
+        thread::spawn(move || {
```

Comment:
I see you have a runtime, any reason not to spawn a task here ?

Reply:
i'm not sure how do I spawn a periodic task without blocking the current thread

Reply:
I see what you mean. I would have to dive into the whole code base, blocking the main thread with a runtime is usually not an issue if you spawn all you tasks before.

Reply:
please see [`3d6501b` (#189)](https://github.com/iotaledger/iota.rs/pull/189/commits/3d6501b82490db40f145b4068b738586c2ef3ca2)